diff --git a/node_modules/@rainbow-me/rainbowkit/dist/components/RainbowKitProvider/RainbowKitChainContext.d.ts b/node_modules/@rainbow-me/rainbowkit/dist/components/RainbowKitProvider/RainbowKitChainContext.d.ts
index b63cc28..1934e4d 100644
--- a/node_modules/@rainbow-me/rainbowkit/dist/components/RainbowKitProvider/RainbowKitChainContext.d.ts
+++ b/node_modules/@rainbow-me/rainbowkit/dist/components/RainbowKitProvider/RainbowKitChainContext.d.ts
@@ -6,9 +6,10 @@ export interface RainbowKitChain extends Chain {
 }
 interface RainbowKitChainProviderProps {
     initialChain?: Chain | number;
+    visibleChains: number[];
     children: ReactNode;
 }
-export declare function RainbowKitChainProvider({ children, initialChain, }: RainbowKitChainProviderProps): React.JSX.Element;
+export declare function RainbowKitChainProvider({ children, initialChain, visibleChains, }: RainbowKitChainProviderProps): React.JSX.Element;
 export declare const useRainbowKitChains: () => RainbowKitChain[];
 export declare const useInitialChainId: () => number | undefined;
 export declare const useRainbowKitChainsById: () => Record<number, RainbowKitChain>;
diff --git a/node_modules/@rainbow-me/rainbowkit/dist/components/RainbowKitProvider/RainbowKitProvider.d.ts b/node_modules/@rainbow-me/rainbowkit/dist/components/RainbowKitProvider/RainbowKitProvider.d.ts
index fe12c34..7fc338d 100644
--- a/node_modules/@rainbow-me/rainbowkit/dist/components/RainbowKitProvider/RainbowKitProvider.d.ts
+++ b/node_modules/@rainbow-me/rainbowkit/dist/components/RainbowKitProvider/RainbowKitProvider.d.ts
@@ -14,6 +14,7 @@ export type Theme = ThemeVars | {
 };
 export interface RainbowKitProviderProps {
     initialChain?: Chain | number;
+    visibleChains?: number[];
     id?: string;
     children: ReactNode;
     theme?: Theme | null;
@@ -28,4 +29,4 @@ export interface RainbowKitProviderProps {
     modalSize?: ModalSizes;
     locale?: Locale;
 }
-export declare function RainbowKitProvider({ appInfo, avatar, children, coolMode, id, initialChain, locale, modalSize, showRecentTransactions, theme, }: RainbowKitProviderProps): React.JSX.Element;
+export declare function RainbowKitProvider({ appInfo, avatar, children, coolMode, id, initialChain, visibleChains, locale, modalSize, showRecentTransactions, theme, }: RainbowKitProviderProps): React.JSX.Element;
diff --git a/node_modules/@rainbow-me/rainbowkit/dist/index.js b/node_modules/@rainbow-me/rainbowkit/dist/index.js
index eab78f6..20c47aa 100644
--- a/node_modules/@rainbow-me/rainbowkit/dist/index.js
+++ b/node_modules/@rainbow-me/rainbowkit/dist/index.js
@@ -883,14 +883,19 @@ var RainbowKitChainContext = createContext4({
 });
 function RainbowKitChainProvider({
   children,
-  initialChain
+  initialChain,
+  visibleChains
 }) {
   const { chains } = useConfig();
+  let availableChains = chains;
+  if (visibleChains.length > 0) {
+    availableChains = availableChains.filter(({ id }) => visibleChains.includes(id));
+  }
   return /* @__PURE__ */ React10.createElement(RainbowKitChainContext.Provider, {
     value: useMemo5(() => ({
-      chains: provideRainbowKitChains(chains),
+      chains: provideRainbowKitChains(availableChains),
       initialChainId: typeof initialChain === "number" ? initialChain : initialChain == null ? void 0 : initialChain.id
-    }), [chains, initialChain])
+    }), [availableChains, initialChain])
   }, children);
 }
 var useRainbowKitChains = () => useContext3(RainbowKitChainContext).chains;
@@ -1999,6 +2004,7 @@ function RainbowKitProvider({
   coolMode = false,
   id,
   initialChain,
+  visibleChains,
   locale,
   modalSize = ModalSizeOptions.WIDE,
   showRecentTransactions = false,
@@ -2017,7 +2023,8 @@ function RainbowKitProvider({
   };
   const avatarContext = avatar != null ? avatar : defaultAvatar;
   return /* @__PURE__ */ React23.createElement(RainbowKitChainProvider, {
-    initialChain
+    initialChain,
+    visibleChains: visibleChains || []
   }, /* @__PURE__ */ React23.createElement(WalletButtonProvider, null, /* @__PURE__ */ React23.createElement(I18nProvider, {
     locale
   }, /* @__PURE__ */ React23.createElement(CoolModeContext.Provider, {
@@ -2670,7 +2677,6 @@ function AccountModal({ onClose, open }) {
 // src/components/ChainModal/ChainModal.tsx
 import React41, { useContext as useContext13, useState as useState12 } from "react";
 import { useAccount as useAccount12, useDisconnect as useDisconnect2, useSwitchChain } from "wagmi";
-import { useConfig as useConfig2 } from "wagmi";
 
 // src/components/Icons/DisconnectSq.tsx
 import React38 from "react";
@@ -2830,7 +2836,6 @@ var MobileScrollClassName = "_18dqw9x1";
 // src/components/ChainModal/ChainModal.tsx
 function ChainModal({ onClose, open }) {
   const { chainId } = useAccount12();
-  const { chains } = useConfig2();
   const [pendingChainId, setPendingChainId] = useState12(null);
   const { switchChain } = useSwitchChain({
     mutation: {
@@ -2854,9 +2859,9 @@ function ChainModal({ onClose, open }) {
   const { disconnect } = useDisconnect2();
   const titleId = "rk_chain_modal_title";
   const mobile = isMobile();
-  const isCurrentChainSupported = chains.some((chain) => chain.id === chainId);
-  const chainIconSize = mobile ? "36" : "28";
   const rainbowkitChains = useRainbowKitChains();
+  const isCurrentChainSupported = rainbowkitChains.some((chain) => chain.id === chainId);
+  const chainIconSize = mobile ? "36" : "28";
   if (!chainId) {
     return null;
   }
